kasa/iot/iotstrip.py

```diff
@@ -176,57 +204,93 @@ async def get_emeter_daily(
+    def has_periodic_stats(self) -> bool:
+        """Return True if device can report statistics for different time periods."""
+        return True
+
+    async def erase_stats(self):
         """Erase energy meter statistics for all plugs."""
-        for plug in self.children:
-            await plug.erase_emeter_stats()
+        for plug in self._device.children:
+            await plug.modules[Module.Energy].erase_stats()
 
     @property  # type: ignore
-    @requires_update
-    def emeter_this_month(self) -> float | None:
+    def consumption_this_month(self) -> float | None:
         """Return this month's energy consumption in kWh."""
-        return sum(v if (v := plug.emeter_this_month) else 0 for plug in self.children)
+        return sum(
+            v if (v := plug.modules[Module.Energy].consumption_this_month) else 0.0
+            for plug in self._device.children
+        )
 
     @property  # type: ignore
-    @requires_update
-    def emeter_today(self) -> float | None:
+    def consumption_today(self) -> float | None:
         """Return this month's energy consumption in kWh."""
-        return sum(v if (v := plug.emeter_today) else 0 for plug in self.children)
+        return sum(
+            v if (v := plug.modules[Module.Energy].consumption_today) else 0.0
+            for plug in self._device.children
+        )
 
     @property  # type: ignore
-    @requires_update
-    def emeter_realtime(self) -> EmeterStatus:
+    def consumption_total(self) -> float | None:
+        """Return total energy consumption since reboot in kWh."""
+        return sum(
+            v if (v := plug.modules[Module.Energy].consumption_total) else 0.0
+            for plug in self._device.children
+        )
+
+    @property  # type: ignore
+    def status(self) -> EmeterStatus:
         """Return current energy readings."""
-        emeter = merge_sums([plug.emeter_realtime for plug in self.children])
+        emeter = merge_sums(
+            [plug.modules[Module.Energy].status for plug in self._device.children]
+        )
         # Voltage is averaged since each read will result
         # in a slightly different voltage since they are not atomic
-        emeter["voltage_mv"] = int(emeter["voltage_mv"] / len(self.children))
+        emeter["voltage_mv"] = int(emeter["voltage_mv"] / len(self._device.children))
         return EmeterStatus(emeter)
 
+    @property
+    def current(self) -> float | None:
+        """Return the current in A."""
+        return self.status.current
+
+    @property
+    def voltage(self) -> float | None:
+        """Get the current voltage in V."""
+        return self.status.voltage
+
+    @property
+    def has_voltage_current(self) -> bool:
+        """Return True if the device reports current and voltage."""
+        return True
+
+    @property
+    def has_total_consumption(self) -> bool:
+        """Return True if device reports total energy consumption since last reboot."""
+        return True
```

Comment:
I'd rather avoid this type of properties, as it reminds me about the `is_*` we used to have. Instead, we'd return `None` on unavailable & handle this through the feature interface?

Reply:
I think we should keep these as it's consistent with the other modules where we have things like `has_custom_effect`, `has_save_preset`, `is_color`, `is_color_temp` etc. It also makes the logic around whether to add the features or not more deterministic than a null check. N.B. This is only on the interface, features will not be created for the `has` attributes, they just won't be created.

Reply:
Replace with `supports` `ModuleFeature` flags